workflow/controller/workflowpod.go

```diff
@@ -249,19 +246,63 @@ func (woc *wfOperationCtx) newWaitContainer(tmpl *wfv1.Template) (*apiv1.Contain
 	ctr := woc.newExecContainer(common.WaitContainerName, false)
 	ctr.Command = []string{"argoexec"}
 	ctr.Args = []string{"wait"}
-	ctr.VolumeMounts = []apiv1.VolumeMount{
+	ctr.VolumeMounts = woc.createVolumeMounts()
+	return ctr, nil
+}
+
+func (woc *wfOperationCtx) createEnvVars() []apiv1.EnvVar {
+	if woc.controller.Config.ContainerRuntimeExecutor == common.ContainerRuntimeExecutorDocker {
```

Comment:
What do you think of set `woc.controller.Config.ContainerRuntimeExecutor`to `common.ContainerRuntimeExecutorDocker` by default around [here](https://github.com/argoproj/argo/blob/b49eeaad87708c78ee396c1b3311af785a78afce/cmd/argoexec/commands/root.go#L85) ?

Reply:
Yea, I like this better. Thanks!

Reply:
To clarify, we would need to replicate switch server side. Because The controller needs to decide how to populate environment variables/volumes depending on the config, and the config can be blank. So something like this would need to be done in the controller:

```
	switch woc.controller.Config.ContainerRuntimeExecutor {
    case common.ContainerRuntimeExecutorKubelet:
        return append(execEnvVars,
            apiv1.EnvVar{
                Name:  common.EnvVarContainerRuntimeExecutor,
                Value: woc.controller.Config.ContainerRuntimeExecutor,
            },
            ...
            ...
    default:
        return execEnvVars
    }
```